worldedit-core/src/main/java/com/sk89q/worldedit/command/GenerationCommands.java

```diff
@@ -154,7 +174,24 @@ public void hsphere(Player player, LocalSession session, EditSession editSession
     )
     @CommandPermissions("worldedit.generation.sphere")
     @Logging(PLACEMENT)
-    public void sphere(Player player, LocalSession session, EditSession editSession, Pattern pattern, String radiusString, @Optional("false") boolean raised, @Switch('h') boolean hollow) throws WorldEditException {
+    public void sphere(Player player, LocalSession session, EditSession editSession, Pattern pattern, String radiusString, @Optional("false") boolean raised, @Switch('h') boolean hollow, @Switch('s') boolean hemi, @Switch('i') boolean upsideDown, @Switch('p') boolean precisionMode) throws WorldEditException {
+        boolean error = false;
+        if(!hemi) {
+            if(precisionMode) {
+                player.printError("-p flag doesn't have effect when it's not an hemisphere.");
```

Comment:
`doesn't have an effect`, not `doesn't have effect`

`not generating a hemisphere`, not `it's not an hemisphere`